cpp/ql/test/library-tests/ir/ssa/ssa.cpp

```diff
@@ -189,4 +189,13 @@ static void AsmStmtWithOutputs(unsigned int& a, unsigned int& b, unsigned int& c
     : "+a" (a), "+b" (b)
     : "c" (c), "d" (d)
     );
+}
+
+int strcmp(const char *, const char *);
+int abs(int);
+
+int PureFunctions(char *str1, char *str2, int x) {
+  int ret = strcmp(str1, str2);
+  ret += abs(x);
+  return ret;
```

Comment:
You wrote in https://github.com/Semmle/ql/pull/1542#issuecomment-509420715 that this PR was about the side effects of `strlen`, so shouldn't that be included in the test?

Reply:
Added.